splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java

```diff
@@ -111,6 +109,42 @@ public void vacuumDatabase() throws SQLException{
                         }
                         continue; //ignore system tables
                     }
+                    boolean requiresDroppedId = false;
+                    if (table.getTransactionId() != null) {
+                        TxnView txn = SIDriver.driver().getTxnSupplier().getTransaction(Long.parseLong(table.getTransactionId()));
+                        if (txn.getEffectiveBeginTimestamp() >= oldestActiveTransaction) {
+                            // This conglomerate was created by a "recent" transaction (newer than the oldest active txn)
+                            // ignore it in case it's still in use
+
+                            if (LOG.isInfoEnabled()) {
+                                LOG.info("Ignoring recently created table: " + table.getTableName() + " by transaction " + txn.getTxnId());
+                            }
+                            continue;
+                        }
+                        if (txn.getEffectiveState().equals(Txn.State.ROLLEDBACK)) {
+                            // Transaction is rolled back, we can remove it safely
+                        } else {
+                            // This conglomerate requires a dropped transaction id
+                            requiresDroppedId = true;
+                        }
+                    }
+                    if (requiresDroppedId) {
+                        if (table.getDroppedTransactionId() != null) {
+                            TxnView txn = SIDriver.driver().getTxnSupplier().getTransaction(Long.parseLong(table.getDroppedTransactionId()));
+                            if (txn.getEffectiveCommitTimestamp() == -1 || txn.getEffectiveCommitTimestamp() >= oldestActiveTransaction) {
+                                // This conglomerate was dropped by an active, rolled back or "recent" transaction
+                                // (newer than the oldest active txn) ignore it in case it's still in use
+
+                                if (LOG.isInfoEnabled()) {
+                                    LOG.info("Ignoring recently dropped table: " + table.getTableName() + " by transaction " + txn.getTxnId());
+                                }
+                                continue;
+                            }
+                        } else {
+                            // This table must have a dropped transaction id before we can process it for vacuum
+                            continue;
+                        }
+                    }
                     if(!activeConglomerates.contains(tableConglom)){
```

Comment:
For old tables without a transaction id and drop transaction id, can vacuum physically drop it? If not, still needs to check sys.sysconglomerate table?

Reply:
Yes, it can. We always check sys.sysconglomerates anyway, the transactionId logic just makes sure that the tables that cannot be found in sys.sysconglomerates are safe to delete.

I have a test case for your scenario in VacuumIT.testVacuumDoesNotRemoveTablesWithoutTransactionId

While looking at this I realized there was a logic gap regarding tables without a transactionId (created before this change goes in) but with droppedTransactionId (deleted after this change). I wasn't checking whether they were still in use. I addressed that and added a test case: testVacuumDoesNotDeleteTableWithoutTransactionIdPossiblyInUse